test/widgets/profile_test.dart

```diff
@@ -0,0 +1,269 @@
+  required int pageUserId,
+  List<User>? users,
+  List<CustomProfileField>? customProfileFields,
+  Map<String, RealmDefaultExternalAccount>? realmDefaultExternalAccounts,
+  NavigatorObserver? navigatorObserver
+}) async {
+  addTearDown(testBinding.reset);
+
+  final initialSnapshot = eg.initialSnapshot(
+    customProfileFields: customProfileFields,
+    realmDefaultExternalAccounts: realmDefaultExternalAccounts);
+  await testBinding.globalStore.add(eg.selfAccount, initialSnapshot);
+  final store = await testBinding.globalStore.perAccount(eg.selfAccount.id);
+
+  store.addUser(eg.selfUser);
+  if (users != null) {
+    store.addUsers(users);
+  }
+
+  await tester.pumpWidget(
+    GlobalStoreWidget(
+      child: MaterialApp(
+        navigatorObservers: navigatorObserver != null ? [navigatorObserver] : [],
+        home: PerAccountStoreWidget(
+          accountId: eg.selfAccount.id,
+          child: ProfilePage(userId: pageUserId)))));
+
+  // global store, per-account store, and page get loaded
+  await tester.pumpAndSettle();
+}
+
+CustomProfileField mkCustomProfileField(int id, CustomProfileFieldType type, {int? order, bool? displayInProfileSummary, String? fieldData}) {
+  return CustomProfileField(
+    id: id,
+    order: order ?? id,
+    name: 'field$id',
+    hint: 'hint$id',
+    displayInProfileSummary: displayInProfileSummary ?? true,
+    fieldData: fieldData ?? '',
+    type: type,
+  );
+}
+
+void main() {
+  TestZulipBinding.ensureInitialized();
+
+  group('ProfilePage', () {
+    testWidgets('page builds; profile page renders', (WidgetTester tester) async {
+      final user = eg.user(userId: 1, fullName: 'test user');
+
+      await setupPage(tester, users:[user], pageUserId: user.userId);
+
+      check(because: 'find user avatar', find.byType(Avatar).evaluate()).length.equals(1);
+      check(because: 'find user name', find.text('test user').evaluate()).isNotEmpty();
+    });
+
+    testWidgets('page builds; profile page renders with profileData', (WidgetTester tester) async {
+      await setupPage(tester,
+        users: [
+          eg.user(userId: 1, profileData: {
+            0: ProfileFieldUserData(value: 'shortTextValue'),
+            1: ProfileFieldUserData(value: 'longTextValue'),
+            2: ProfileFieldUserData(value: 'x'),
+            3: ProfileFieldUserData(value: 'dateValue'),
+            4: ProfileFieldUserData(value: 'http://example/linkValue'),
+            5: ProfileFieldUserData(value: '[2]'),
+            6: ProfileFieldUserData(value: 'externalValue'),
+            7: ProfileFieldUserData(value: 'pronounsValue'),
+          }),
+          eg.user(userId: 2),
+        ],
+        pageUserId: 1,
+        customProfileFields: [
+          mkCustomProfileField(0, CustomProfileFieldType.shortText),
+          mkCustomProfileField(1, CustomProfileFieldType.longText),
+          mkCustomProfileField(2, CustomProfileFieldType.choice, fieldData: '{"x": {"text": "choiceValue", "order": "1"}}'),
```

Comment:
nit:
```suggestion
          mkCustomProfileField(2, CustomProfileFieldType.choice,
            fieldData: '{"x": {"text": "choiceValue", "order": "1"}}'),
```

(and similarly the `externalAccount` line below)

We don't adhere to a strict line-length limit like 80 columns. But I do find it quite valuable to have two source files on the screen at the same time side by side, and on a wide screen also a terminal and a browser (for docs, or for a GitHub review page), so keeping a lid on the code's width is needed for that.

In zulip-mobile we had Prettier apply a hard limit of 100 columns, and that worked reasonably well — partly because the nature of things is that most code doesn't approach the limit, so I could set a viewport to more like 90 columns and still have only occasional wrapping or need to scroll. We've generally been going for similar widths here in zulip-flutter, which might be summarized as:
* up to 80 is always fine
  * (well, except for prose, which is a different kettle of fish)
* up to about 90 is fine if it makes the formatting observably nicer even in a small way
* up to about 100 is fine if the alternative would be a lot more ugly, and if the text after column 90 or so is boring (so that the reader can largely get away with not scrolling right to read it)
* much more than 100 should almost always be reformatted or otherwise rearranged
* URLs are an exception — can't break those up without breaking them
* sometimes also bend this for the sake of greppability, like putting a comment that has a grep-friendly TODO(something) marker on the same line as the code it refers to (and this last point is a change from zulip-mobile, because there's no good way to teach Prettier about it).

This line is 124 columns wide, and the other one below is 113, so well past those widths.

Reply:
I was following the guideline in https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#prefer-a-maximum-line-length-of-80-characters which has a (quite extra) example that seemed to argue for me leaving it as is, but it does make sense given these two fields have fieldData and actually not consistent with the other lines (so the difference does convey information)

Reply:
Yeah, in their example:
```dart
// BETTER (consistent lines, not much longer than the earlier example)
final int a = 1;
final int b = 2;
final int c = a.very.very.very.very.very.long.expression.that.returns.three.eventually().but.is.very.long();
final int d = 4;
final int e = 5;
```

I would break up the expression:
```dart
final int a = 1;
final int b = 2;
final int c = a.very.very.very.very.very.long.expression.that
  .returns.three.eventually().but.is.very.long();
final int d = 4;
final int e = 5;
```

I do sympathize with the principle they're getting at, that consistency in a list is better than sticking to a rigid width limit. And I agree that that "BETTER" form is better than the "BAD" form of the same example:
```dart
// BAD (breaks after assignment operator and still goes over 80 chars)
final int a = 1;
final int b = 2;
final int c =
    a.very.very.very.very.very.long.expression.that.returns.three.eventually().but.is.very.long();
final int d = 4;
final int e = 5;
```
because it's not that much wider and is significantly more regular.

Here, I think this is a key aspect of it:
> it does make sense given these two fields have fieldData and actually not consistent with the other lines (so the difference does convey information)